### PR TITLE
Numericcategories routes

### PR DESCRIPTION
**Pull request by karishma-gangwani**
## Description
All tests pass

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/routes/termdb.getpercentile.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// import { getpercentileRequest, getpercentileResponse } from '#shared/types/routes/termdb.getpercentile'
+import { getpercentileRequest, getpercentileResponse } from '#shared/types/routes/termdb.getpercentile.ts'
 import * as termdbsql from '#src/termdb.sql.js'
 import computePercentile from '#shared/compute.percentile.js'
+import { Filter } from 'shared/types'
```
Comment (xzhou82) [inline server/routes/termdb.getpercentile.ts:4]: typo, `#shared/`

File: server/routes/termdb.categories.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ function init({ genomes }) {
 
 async function trigger_getcategories(
 	q: { tid: string | number; type: string; filter: any; term1_q: any; currentGeneNames?: string[]; rglst?: any },
-	res: { send: (arg0: { lst: any[]; orderedLabels: any }) => void },
+	res: { send: (arg0: { lst: any[]; orderedLabels?: any }) => void },
```
Comment (xzhou82) [inline server/routes/termdb.categories.ts:92]: send() parameter type should be `getcategoriesResponse`

File: server/shared/types/routes/termdb.getnumericcategories.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Filter } from '../filter'
+
+export type getnumericcategoriesRequest = {
+	/** a user-defined genome label in the serverconfig.json, hg38, hg19, mm10, etc */
+	genome: string
+	/** a user-defined dataset label in the serverconfig.json, such as ClinVar, SJLife, GDC, etc */
+	dslabel: string
+	embedder: string
+	getnumericcategories: number
```
Comment (xzhou82) [inline server/shared/types/routes/termdb.getnumericcategories.ts:9]: this property is no longer needed, please delete here and also from vocab `getNumericUncomputableCategories()` arg

**Comment by gavrielm** [comment]
There is considerable code duplication between `server/routes/termdb.getdescrstats.ts` and `client/shared/descriptive.stats.js`. May consider merging the code and only using the server route file.

**Review by xzhou82 — APPROVED** [review]
works thanks. next `termdb.getdescrstats.ts` can import code from '#shared/descriptive.stats.js`

**Pull request merged**